tuned_lens/plotting/trajectory_plotting.py

```diff
@@ -1,24 +1,71 @@
 """Contains utility classes for creating heatmap visualizations."""
-from dataclasses import dataclass
-from typing import Any, Dict, Optional
+from dataclasses import dataclass, replace
+from typing import Any, Dict, Optional, Tuple
 
 import numpy as np
 from numpy.typing import NDArray
 from plotly import graph_objects as go
 
 
+def trunc_string_left(string: str, new_len: int) -> str:
+    """Truncate a string to the left."""
+    return " " * (new_len - len(string)) + string[-new_len:]
+
+
 @dataclass
 class TrajectoryLabels:
     """Contains sets of labels for each layer and position in the residual stream."""
 
     # (n_layers x sequence_length) label for each layer and position in the stream.
     label_strings: NDArray[np.str_]
-    # (sequence_length) labels for the sequence dimension typically the input tokens.
-    sequence_labels: NDArray[np.str_]
-    # (n_layers x sequence_length x k) k entries to display when hovering over a cell.
-    # For example, the top k prediction from the lens at each layer.
+    # (n_layers x sequence_length x rows x cols) table of strings to display when
+    # hovering over a cell. For example, the top k prediction from the lens.
     hover_over_entries: Optional[NDArray[np.str_]] = None
 
+    def stride(self, stride: int) -> "TrajectoryLabels":
```

Comment:
This function and `template_and_customdata` don't seem to be exercised by the tests. Are they in one of the notebooks above?

Reply:
They are but I also added a unit test